storage/remote/storage.go

```diff
@@ -65,12 +65,16 @@ type Storage struct {
 
 // NewStorage returns a remote.Storage.
 func NewStorage(l *slog.Logger, reg prometheus.Registerer, stCallback startTimeCallback, walDir string, flushDeadline time.Duration, sm ReadyScrapeManager) *Storage {
+	// register remote storage metrics in	custom registry
+	if reg != nil {
+		reg.MustRegister(samplesIn, histogramsIn, exemplarsIn)
```

Comment:
Same here - do we need to unregister?

Reply:
I think this is the same as above. Here we are already dealing with a custom registry and are registering other metrics with it without worrying about unregistration. There is only ever one storage, so that's fine for now. (Should somebody need deregistration one day, we would need to add it.)